docs/turbo-native-modules-ios.md

```diff
@@ -0,0 +1,129 @@
+
+1. Open the CocoPods generated Xcode Workspace:
+
+```bash
+cd ios
+open TurboModuleExample.xcworkspace
+```
+
+<img class="half-size" alt="Open Xcode Workspace" src="/docs/assets/turbo-native-modules/xcode/1.webp" />
+
+2. Right click on app and select <code>New Group</code>, call the new group `NativeLocalStorage`.
+
+<img class="half-size" alt="Right click on app and select New Group" src="/docs/assets/turbo-native-modules/xcode/2.webp" />
+
+3. In the `NativeLocalStorage` group, create <code>New</code>→<code>File from Template</code>.
+
+<img class="half-size" alt="Create a new file using the Cocoa Touch Classs template" src="/docs/assets/turbo-native-modules/xcode/3.webp" />
+
+4. Use the <code>Cocoa Touch Class</code>
+
+<img class="half-size" alt="Use the Cocoa Touch Class template" src="/docs/assets/turbo-native-modules/xcode/4.webp"  />
+
+5. Name the Cocoa Touch Class <code>RCTNativeLocalStorage</code> with the <code>Objective-C</code> language
+
+<img class="half-size" alt="Create an Objective-C RCTNativeLocalStorage class" src="/docs/assets/turbo-native-modules/xcode/4.webp" />
+
+6. Rename <code>RCTNativeLocalStorage.m</code> → <code>RCTNativeLocalStorage.mm</code> making it an Objective-C++ file
+
+<img class="half-size" alt="Convert to and Objective-C++ file" src="/docs/assets/turbo-native-modules/xcode/5.webp" />
+
+## Implement localStorage with NSUserDefaults
+
+Start by updating `RCTNativeLocalStorage.h`:
+
+```objc
+//  RCTNativeLocalStorage.h
+//  TurboModuleExample
+
+#import <Foundation/Foundation.h>
+// highlight-add-next-line
+#import <NativeLocalStorageSpec/NativeLocalStorageSpec.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// highlight-remove-next-line
+@interface RCTNativeLocalStorage : NSObject
+// highlight-add-next-line
+@interface RCTNativeLocalStorage : NSObject <NativeLocalStorageSpec>
+
+@end
+```
+
+Then update our implementation to use `NSUserDefaults.
+
+```objc
+//  RCTNativeLocalStorage.mm
+//  TurboModuleExample
+
+#import "RCTNativeLocalStorage.h"
+
+@implementation RCTNativeLocalStorage
+
+// highlight-add-start
++RCT_EXPORT_MODULE(NativeLocalStorage)
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const facebook::react::ObjCTurboModule::InitParams &)params {
+  return std::make_shared<facebook::react::NativeLocalStorageSpecJSI>(params);
+}
+
+- (NSString * _Nullable)getItem:(NSString *)key {
+  return [NSUserDefaults.standardUserDefaults stringForKey:key];
+}
+
+- (void)setItem:(NSString *)value
+          key:(NSString *)key {
+  [NSUserDefaults.standardUserDefaults setObject:value forKey:key];
+}
+
+- (void)removeItem:(NSString *)key {
+  [NSUserDefaults.standardUserDefaults removeObjectForKey:key];
+}
+
+- (void)clear {
+  NSDictionary *keys = [NSUserDefaults.standardUserDefaults dictionaryRepresentation];
+  for (NSString *key in keys) {
+    [self removeItem:key];
+  }
```

Comment:
this is dangerous. The standard default local storage stores also other keys, like the lunching arguments and some default settings for the app. If we want to implement the clear funcion, we should not use the `standardUserDefaults`, but we need to create a custom one with `[NSUserDefaults new]`.